internal/component/prometheus/exporter/blackbox/blackbox.go

```diff
@@ -93,8 +98,49 @@ func (t TargetBlock) Convert() []blackbox_exporter.BlackboxTarget {
 type Arguments struct {
 	ConfigFile         string                    `alloy:"config_file,attr,optional"`
 	Config             alloytypes.OptionalSecret `alloy:"config,attr,optional"`
-	Targets            TargetBlock               `alloy:"target,block"`
+	Targets            TargetBlock               `alloy:"target,block,optional"`
 	ProbeTimeoutOffset time.Duration             `alloy:"probe_timeout_offset,attr,optional"`
+
+	// New way of passing targets. This allows the component to receive targets from other components.
+	TargetsList TargetsList `alloy:"targets,attr,optional"`
+}
+
+type TargetsList []map[string]string
+
+func (t TargetsList) Convert() []blackbox_exporter.BlackboxTarget {
+	targets := make([]blackbox_exporter.BlackboxTarget, 0, len(t))
+	for _, target := range t {
+		address, _ := getAddress(target)
+		targets = append(targets, blackbox_exporter.BlackboxTarget{
+			Name:   target["name"],
+			Target: address,
+			Module: target["module"],
+		})
+	}
+	return targets
+}
+
+func (t TargetsList) convert() []BlackboxTarget {
```

Comment:
Having `Convert` and `convert` on the same type with different returns is confusing here. Can we rename this one?

Reply:
yeah that's a bit annoying because we have several backbox target types. I renamed it to convertInternal. Not super happy with the name but it's better than having convert twice. Do you have a suggestion or is this ok for you?